rpc/lib/client/http_client.go

```diff
@@ -136,6 +197,57 @@ func (c *JSONRPCClient) SetCodec(cdc *amino.Codec) {
 
 //-------------------------------------------------------------
 
+// Count returns the number of enqueued requests waiting to be sent.
+func (b *JSONRPCRequestBatch) Count() int {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+	return len(b.requests)
+}
+
+func (b *JSONRPCRequestBatch) enqueue(req *JSONRPCBufferedRequest) {
+	b.mtx.Lock()
+	b.requests = append(b.requests, req)
+	b.mtx.Unlock()
+}
+
+// Clear empties out the request batch.
+func (b *JSONRPCRequestBatch) Clear() int {
+	b.mtx.Lock()
+	defer b.mtx.Unlock()
+	return b.clear()
+}
+
+func (b *JSONRPCRequestBatch) clear() int {
+	count := len(b.requests)
+	b.requests = make([]*JSONRPCBufferedRequest, 0)
+	return count
+}
+
+// Send will attempt to send the current batch of enqueued requests, and then
+// will clear out the requests once done. On success, this returns the
+// deserialized list of results from each of the enqueued requests.
+func (b *JSONRPCRequestBatch) Send() ([]interface{}, error) {
+	b.mtx.Lock()
+	defer func() {
+		b.clear()
+		b.mtx.Unlock()
+	}()
+	return b.client.sendBatch(b.requests)
+}
+
+// Call enqueues a request to call the given RPC method with the specified
+// parameters, in the same way that the `JSONRPCClient.Call` function would.
+func (b *JSONRPCRequestBatch) Call(method string, params map[string]interface{}, result interface{}) (interface{}, error) {
+	request, err := types.MapToRequest(b.client.cdc, types.JSONRPCStringID("jsonrpc-client"), method, params)
```

Comment:
Should we make "jsonrpc-client" a field in RPCClient? and append some nonce for every method call?

Reply:
Not sure I understand - is this to make each batched request's response uniquely identifiable and able to be matched to a corresponding request?

Reply:
Better identify it on the server? I don't remember what ID should be according to JSONRPC spec, but I think it should be unique per client, so we should probably use cmn.RandStr? to generate suffix or something

Reply:
Oh so you mean that the Tendermint server can identify different clients? So yes, from the JSON RPC spec:

> The Server MUST reply with the same value in the Response object if included. This member is used to correlate the context between the two objects.

So I don't see any context being tracked on the Tendermint side at the moment for RPC clients (apart from logging with the ID), but should we need that in future it'd be critical to identify clients from each other.

Busy adding this functionality now 👍

Reply:
Done adding this now 👍